mysql-test/suite/binlog_encryption/t/binlog_encryption_without_keyring.test

```diff
@@ -0,0 +1,48 @@
+# Bug: PS-288: mysqld got signal 6 if you try to enable encrypt_binlog + binary log without keyring-file plugin
+
+# Server is aborted when binlog_error_action is set to ABORT_SERVER and binlog encryption could not be
+# initialized. An error message generated in server log got improved in this case by mentioning that
+# encryption could be a reason why server got aborted.
+
+--source include/not_embedded.inc
+--source include/have_log_bin.inc
+
+--let restart_hide_args=1
+--let $restart_parameters=restart:--log-error=$MYSQLTEST_VARDIR/tmp/binlog_encryption_without_keyring.err
+--source include/restart_mysqld.inc
+
+UNINSTALL PLUGIN keyring_file;
+
+--echo # Make sure that no keyring plugin is installed
+query_vertical SELECT PLUGIN_NAME,PLUGIN_VERSION,PLUGIN_STATUS
+FROM INFORMATION_SCHEMA.PLUGINS WHERE plugin_name LIKE 'keyring%';
+
+--let $assert_text= Binlog should be ON
+--let $assert_cond= "[SELECT @@GLOBAL.log_bin = 1]" = 1
+--source include/assert.inc
+
+--let $assert_text= Binlog encryption should be ON
+--let $assert_cond= "[SELECT @@GLOBAL.encrypt_binlog = 1]" = 1
+--source include/assert.inc
+
+--let $saved_binlog_error_action=`SELECT @@GLOBAL.binlog_error_action`
+SET GLOBAL binlog_error_action= ABORT_SERVER;
+--exec echo "wait" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--disable_reconnect
+--echo # open new binlog file
+--error ER_BINLOG_LOGGING_IMPOSSIBLE
+RESET MASTER;
+--echo # Check that error messages related to encryption are present in error log
+--let SEARCH_FILE= $MYSQLTEST_VARDIR/tmp/binlog_encryption_without_keyring.err
+--let SEARCH_PATTERN= Binary logging not possible. Message: Either disk is full or file system is read only or encryption failed while opening the binlog. Aborting the server.
+--source include/search_pattern_in_file.inc
+--let SEARCH_PATTERN= Failed to fetch percona_binlog key from keyring and thus failed to initialize binlog encryption. Have you enabled keyring plugin?
+--source include/search_pattern_in_file.inc
+
+--let $restart_parameters=
+--let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--source include/start_mysqld.inc
+
+eval SET GLOBAL binlog_error_action= $saved_binlog_error_action;
```

Comment:
Saving/restoring is not required across restarts here

Reply:
right